admin_ui/src/components/CSVButton.vue

```diff
@@ -0,0 +1,94 @@
+<template>
+    <a class="button">
+        <downloadexcel
+            :before-finish="finishDownload"
+            :before-generate="startDownload"
+            :fetch="fetchExportedRows"
+            :name="tableName + '.csv'"
+            type="csv"
+        >
+            <span>
+                <font-awesome-icon icon="file-csv" />Export as CSV
+            </span>
+        </downloadexcel>
+    </a>
+</template>
+
+<script lang="ts">
+import axios from "axios"
+import downloadexcel from "vue-json-excel"
+import * as i from "../interfaces"
+
+export default {
+    props: ["tableName"],
+    data() {
+        return {
+            tableName: this.tableName,
+        }
+    },
+    components: {
+        downloadexcel,
+    },
+    methods: {
+        // Export data as csv from json:
+        async fetchExportedRows() {
+            const params = this.$store.state.filterParams
+            const tableName = this.$store.state.currentTableName
+            const sortBy = this.$store.state.sortBy
+            if (sortBy) {
+                let prefix = sortBy.ascending ? "" : "-"
+                params["__order"] = prefix + sortBy.property
+            }
+            // Get the row counts:
+            const response = await axios.get(`api/tables/${tableName}/count/`, {
+                params,
+            })
+            const data = response.data as i.RowCountAPIResponse
+            const localParams = { ...params }
+
+            localParams["__page"] = data.count
+            // Set higher __page_size param to have fewer requests to api:
+            localParams["__page_size"] = 1000
+            const pages = Math.ceil(data.count / localParams["__page_size"])
+            const exportedRows = []
+
+            try {
+                for (let i = 1; i < pages + 1; i++) {
+                    localParams["__page"] = i
+                    const response = await axios.get(
+                        `api/tables/${tableName}/?__readable=true`,
+                        {
+                            params: localParams,
+                        }
+                    )
+                    exportedRows.push(...response.data.rows)
+                }
+                return exportedRows
+            } catch (error) {
+                console.log(error.response)
+            }
+        },
+        async startDownload() {
+            const params = this.$store.state.filterParams
+            const tableName = this.$store.state.currentTableName
+            const response = await axios.get(`api/tables/${tableName}/count/`, {
+                params,
+            })
+            const data = response.data as i.RowCountAPIResponse
```

Comment:
I just realised, the row count is available in the store as `this.$store.state.rowCount`. It's calculated when the page loads, so if the page has been open a while, it could be slightly out of date, but probably still OK to use.

If we decide to keep with the current approach, and fetch the latest count, it might be worth storing the count as a data attribute - i.e. `this.rowCount = response.count`, so it can be reused between `startDownload` and `fetchExportedRows`. It just saves one API call.